include/swift/AST/DiagnosticsSema.def

```diff
@@ -360,6 +360,11 @@ ERROR(cannot_convert_argument_value_generic,none,
       "cannot convert value of type %0 (%1) to expected argument type %2 (%3)",
       (Type, StringRef, Type, StringRef))
 
+ERROR(ambiguous_generic_type_inference, none,
+      "cannot infer type for to generic parameter %0; unable to disambiguate "
```

Comment:
This is technically incorrect - solver did infer generic parameter to some time, it's just that it so happened that it was N different types... I think existing `conflicting_arguments_for_generic_parameter` fits well with this problem.

Reply:
> This is technically incorrect - solver did infer generic parameter to some time, it's just that it so happened that it was N different types... 

Ahh, true... the intention was just say that it cannot disambiguate

>  I think existing conflicting_arguments_for_generic_parameter fits well with this problem.

I used that at first, but found a bit off for this case because of 'conflicting arguments to generic' but in fact there is just one argument on the function and the other binding comes from the result type being infered form the context of apply to arg to the outter call to `SR9839` not `id` arguments. Interpreting arguments as `id` arguments :)

Reply:
In this case it doesn't really matter where the binding comes from, what matter is that conflict is in generic parameter `T`.